plugins/services/src/js/components/forms/ContainerServiceFormAdvancedSection.tsx

```diff
@@ -268,6 +272,96 @@ class ContainerServiceFormAdvancedSection extends React.Component {
             <FieldError>{diskErrors}</FieldError>
           </FormGroup>
         </FormRow>
+        <Trans render="h2" className="short-bottom">
+          Limits
+        </Trans>
+        <Trans render="p">Limits settings for cpu and mem.</Trans>
+        <FormRow>
+          <FormGroup className="column-4" showError={Boolean(limitsErrors)}>
+            <FieldLabel className="text-no-transform">
+              <FormGroupHeading>
+                <FormGroupHeadingContent>
+                  <Trans render="span">CPUs</Trans>
+                </FormGroupHeadingContent>
+              </FormGroupHeading>
+            </FieldLabel>
+            <FieldInput
+              min="0"
+              name="limits.cpus"
+              step="0.01"
+              type="number"
+              value={
+                findNestedPropertyInObject(data, limitsPath + ".cpus.value") ||
+                (findNestedPropertyInObject(
```

Comment:
there MUST be a way to make this more intention revealing. took me like a minute to understand that all deez lines just handle the case there's a `0`.

Reply:
Changed this to: 
```JS
data?.limits?.cpus?.value ||
(data?.limits?.cpus?.value === 0 ? 0 : "")
```

This should be nicer to read

Reply:
would `data?.limits?.cpus?.value ?? ""` work?

Reply:
Yes I did not know that there is null coalescing in typescript 🆒 😎 will change.